### DISCUSSION TITLE
How can I use telegram bot packages in Next.js without custom server

### DISCUSSION THREAD

**Discussion by sekiranat** [vercel/next.js#60379] [Help]
I can use telegraf.js (or others) bot using middleware in express, but can't with next.js due to runtime edge in middlware.
Such packages can be integrated into an existing web server port using a middleware:

```js
const bot = new Telegraf(process.env.BOT_TOKEN)
expressApp.use(bot.webhookCallback('/secret-path'))
bot.telegram.setWebhook(`${myDomen}/secret-path`)
```

Because in next.js there is no node runtime in middleware I can't go the same way.
Is it possible to use a bot packages without a custom server + nextJs?

**Comment by rogeliorv** [comment]
Hi @sekiranat 

In this case you would need to configure telegram webhooks to hit your NextJS /api endpoints.

I had a similar question as you and all the information I ran into required a long running process in nodejs using the Telegram polling mechanisms.

Take a look here: https://github.com/rogeliorv/teletubby

**Comment by rishi-raj-jain** [comment]
@sekiranat 

I believe you can use Grammy SDK with Next.js App Router to achieve your end goal.

Take a look here: https://www.launchfa.st/blog/telegram-nextjs-app-router
